cmd/node/node.go

```diff
@@ -524,9 +524,10 @@ func (app *App) initServices(ctx context.Context,
 		return errors.New("invalid golden atx id")
 	}
 
-	beaconProtocol := beacon.New(nodeID, app.host, sgn, vrfSigner, cdb, clock,
+	beaconProtocol := beacon.New(nodeID, sgn, vrfSigner, cdb, clock,
 		beacon.WithContext(ctx),
 		beacon.WithConfig(app.Config.Beacon),
+		beacon.WithPublisher(app.host),
```

Comment:
i think options are supposed to be used for optional params, such as logger, config, context

publisher is not optional, as it will be nil if not set